tools/image-tool/src/crane.rs

```diff
@@ -0,0 +1,41 @@
+use std::path::Path;
+
+use async_trait::async_trait;
+
+use crate::{cli::CommandLine, ImageTool, Result};
+
+pub struct CraneCLI {
+    pub(crate) cli: CommandLine,
+}
+
+#[async_trait]
+impl ImageTool for CraneCLI {
+    async fn pull_archive(&self, path: &Path, uri: &str, _cache: &Path) -> Result<()> {
+        let archive_path = path.to_string_lossy();
+        self.cli
+            .spawn(
+                &["pull", "--format", "oci", uri, archive_path.as_ref()],
+                format!("failed to pull image archive from {}", uri),
+            )
+            .await?;
+        Ok(())
+    }
+
+    async fn get_manifest(&self, uri: &str) -> Result<Vec<u8>> {
+        self.cli
+            .output(
+                &["manifest", uri],
+                format!("failed to fetch manifest for resource at {}", uri),
+            )
+            .await
+    }
+
+    async fn get_config(&self, uri: &str) -> Result<Vec<u8>> {
```

Comment:
as neat as this particular function is I think it may make more sense for this interface to be `get_image_labels` instead of `get_config`. the `docker` and `crane` implementations of this are returning different JSON objects that just happen to both contain a field named either "config"/"Config" that also happen to contain a field called "Labels". IMO the sort-of duplicated code is more robust than having these impls return completely different JSON strings.

Reply:
Talked offline, we landed on a compromise of returning the actual Config object that is shared between both data types and is consistent